### PR TITLE
add analytic content for Digital Guardian

### PR DESCRIPTION
**Pull request by spsocprime**
## Proposed Changes

Analytic content for Digital Guardian

### CODE REVIEW

**Comment by v-sabiraj** [comment]
The name 'DigitalGuardianDLPEvent' does not refer to any known table, please add  'DigitalGuardianDLPEvent' to https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

**Comment by v-sabiraj** [comment]
image
Workbooks queries are failing. Please look into it.

**Comment by spsocprime** [comment]
> image Workbooks queries are failing. Please look into it.

Queries are failing because you do not have DG DLP logs and some fields, (e.g timestamp) are extracted from logs data by parser function

**Comment by spsocprime** [comment]
fixed

**Comment by v-sabiraj** [comment]
@aprakash13, Please review Analytic Rules, thanks.

**Comment by v-sabiraj** [comment]
image
image
The workbooks query is failing, please look into it.

**Comment by spsocprime** [comment]
> image image The workbooks query is failing, please look into it.

this query fails because http_url field is probably empty in all events

**Comment by v-sabiraj** [comment]
image
Please fix queries, thanks.

**Comment by v-sabiraj** [comment]
Please fix the error "Where" to "where", the letter 'w' must be in lowercase in hunting queries, thanks.

**Comment by v-sabiraj** [comment]
image
Please fix the query, "Where" to "where", the letter 'w' must be in lowercase, thanks.

**Comment by v-sabiraj** [comment]
image
Please fix the queries for the Analytic Rules with reference to image, Thanks.

**Comment by v-sabiraj** [comment]
> > image image The workbooks query is failing, please look into it.
> 
> this query fails because http_url field is probably empty in all events

Can you please fix this error, from your end?

**Comment by spsocprime** [comment]
> > > image image The workbooks query is failing, please look into it.
> > 
> > 
> > this query fails because http_url field is probably empty in all events
> 
> Can you please fix this error, from your end?

This field is dynamically extracted from specific DG events by the parser. If there are no such events in your environment then Sentinel drops error.

**Comment by v-sabiraj** [comment]
Workbooks looks good, @aprakash13, please review and approve Analytic Rules, thanks.

**Review by v-sabiraj — DISMISSED** [review]

**Comment by v-sabiraj** [comment]
Workbooks looks good, can merge once @aprakash13 review and approve, thanks.

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternalDomain.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  let corp_domain = dynamic(['example.com']);
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternalDomain.yaml:19]: Please add a comment to let users know to add their own domains.

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  let threshold = 10;
+  let corp_domain = dynamic(['example.com']);
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml:20]: Please add a comment to let users know to add their own domains.

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianPossibleProtocolAbuse.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+id: a374a933-f6c4-4200-8682-70402a9054dd
+name: Digital Guardian - Possible protocol abuse
+description: |
+  'Detects possible protocol abuse.'
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianPossibleProtocolAbuse.yaml:4]: Adding SMTP in the description will be better.

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianUnexpectedProtocol.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+id: a14f2f95-bbd2-4036-ad59-e3aff132b296
+name: Digital Guardian - Unexpected protocol
+description: |
+  'Detects unexpected protocol usage.'
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianUnexpectedProtocol.yaml:4]: Might be a good idea to give more detail/context in description regarding why the use of Terminal service/RDP is unexpected.

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - DigitalGuardianDLPEvent
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | where TimeGenerated > ago(24h)
+  | summarize count() by DstIpAddr, DstPortNumber
+  | order by count_ asc
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml:18]: It seems this query will list all the ports. How are we determining which are rare. Wondering if we wanted to use
 top 5 or top 10 here.

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | where TimeGenerated > ago(24h)
+  | where isnotempty(NetworkApplicationProtocol)
+  | summarize count() by SrcIpAddr, SrcUserName
+  | order by count_ asc
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml:19]: How are we determining which protocols are rare. Wondering if we wanted to use
 top 5 or top 10 here.

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareUrls.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | where TimeGenerated > ago(24h)
+  | where isnotempty(http_url)
+  | summarize count() by SrcUserName, http_url
+  | order by count_ asc
```
Comment (aprakash13) [inline Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareUrls.yaml:19]: Wondering if we wanted to use top 5 or top 10 here to present rare URL.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Made a few suggestions. Please take a look and see if the query needs to be updated.

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  - connectorId: DigitalGuardianDLP
+    dataTypes:
+      - DigitalGuardianDLPEvent
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | Where TimeGenerated > ago(24h)
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml]: fixed

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  - connectorId: DigitalGuardianDLP
+    dataTypes:
+      - DigitalGuardianDLPEvent
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | Where TimeGenerated > ago(24h)
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml]: fixed

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternal.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      - DigitalGuardianDLPEvent
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternal.yaml:18]: fiex

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  lett threshold = 10;
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml]: fixed

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianInsecureProtocolSources.yaml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  - connectorId: DigitalGuardianDLP
+    dataTypes:
+      - DigitalGuardianDLPEvent
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048
+query: |
+  DigitalGuardianDLPEvent
+  | Where TimeGenerated > ago(24h)
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianInsecureProtocolSources.yaml]: fixed

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternalDomain.yaml (hunk 1 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFileSentToExternalDomain.yaml]: added comment

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml (hunk 2 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianFilesSentToExternalDomain.yaml:20]: added comment

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianPossibleProtocolAbuse.yaml (hunk 3 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianPossibleProtocolAbuse.yaml]: added protocol name

File: Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianUnexpectedProtocol.yaml (hunk 4 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Analytic Rules/DigitalGuardianUnexpectedProtocol.yaml]: changed description

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml (hunk 5 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareDestinationPorts.yaml:18]: fixed

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml (hunk 6 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareNetworkProtocols.yaml:19]: fixed

File: Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareUrls.yaml (hunk 7 above)
Comment (spsocprime) [reply Solutions/DigitalGuardianDLP/Hunting Queries/DigitalGuardianRareUrls.yaml:19]: fixed

**Review by aprakash13 — APPROVED** [review]
Suggested changes have been fixed. Approved.

**Pull request merged**